### ISSUE TITLE
Reorganize files into logical folders

### ISSUE THREAD

**Issue by xPaw** [SteamDatabase/ValveKeyValue#8]
Currently all VKV files are in a single folder, and it's hard to follow.
Split it into readers, writers, helpers, serializers, etc.

**Comment by yaakov-h** [comment]
Agreed. I was intending to do this once I started on KV3.

I'm somewhat tempted also to have a different test assembly for each KV implementation. What are your thoughts on that?

**Comment by xPaw** [comment]
I don't see the need in having separate assemblies for it.

**Comment by yaakov-h** [comment]
I figured it would be easier to run just the kv1 tests or just the kv2 tests, but I just realised I could use NUnit categories for that so yeah no need.

**Comment by SinZ163** [comment]
0f0021ece96b4031ffa9f6d6fe4bd804ef344959 split the files logically between serializers and deserializers

**Comment by xPaw** [comment]
I moved KV1 specific files into separate folders while I was working on KV3.
